internal/util/pkgutil/pkgutil.go

```diff
@@ -63,10 +63,15 @@ func WalkPackage(src string, c func(string, os.FileInfo, error) error) error {
 	})
 }
 
-// CopyPackage copies the content of a single package from src to dst. It
-// will not copy resources belonging to any subpackages.
-func CopyPackage(src, dst string, copyRootKptfile bool) error {
-	return WalkPackage(src, func(path string, info os.FileInfo, err error) error {
+// CopyPackage copies the content of a single package from src to dst. If includeSubpackages
+// is true, it will copy resources belonging to any subpackages.
+func CopyPackage(src, dst string, copyRootKptfile bool, matcher pkg.SubpackageMatcher) error {
+	subpackagesToCopy, err := pkg.Subpackages(src, matcher, true)
```

Comment:
How does this handle deeply nested subpackages? Remember there might be several layers of subpackages, both local and remote.

Reply:
I was under the impression that local packages (whatever that would mean) under remote subpackages should be handled as if the entirety was a remote subpackage, which this should hopefully do (will address in a test case as mentioned below). Is that not the case?